src/main/java/org/testng/internal/annotations/TestAnnotation.java

```diff
@@ -168,21 +168,21 @@ public void setSequential(boolean sequential) {
 
   @Override
   public IRetryAnalyzer getRetryAnalyzer() {

```

Comment:
I don't catch the change here: a new instance will be created for each get.
And set should throw an exception if the provided class is not a retry analyzer.

Reply:
@juherr - A new instance would be created for each get, but a `get` is invoked only when a new `TestNGMethod` object is created. So we are good there. 
I am not sure as to how would a user get to the part of setting an invalid class via the setter even when using the TestNG APIs. So I chose to skip the exception part. We take in whatever class the user provides, but we instantiate it only if its of type `IRetryAnalyzer` and in all other cases, we return a `null`.

Reply:
It is error-prone because it is currently right but it can change.
Same thing for the null instead of exception: it will silently fail.

Reply:
@juherr 

> It is error-prone because it is currently right but it can change.
Fair enough. I decided to tackle this differently. I left the functionality as is, but modified `TestNGMethod` to create a new instance based on the instance that we get from the actual annotation and work with that instead. Its like cloning, but I just grab hold of the class and use reflection to create a new instance.

> Same thing for the null instead of exception: it will silently fail.
`null` instead of exception piece is basically what is the current behavior as well. Since we cannot possibly inject a non-adhering implementation for retry analyzer, I chose to retain the behavior as is.